### ISSUE TITLE
What would make this more convenient  - Copy  with ctrl +c, ctrl v, trim strings and other things 

### ISSUE THREAD

**Issue by smer44** [atenfyr/UAssetGUI#31]
A huge inconvenience of this tool is that edition is not well done in compare to other text editors.
So pretty trivial suggestion would be, sorted by size:
- possibility to copy/paste text by ctrl c ctrl v via normal operating system buffer
- please trim string input fields, because it may go wrong if you have spaces or newlines at the begin or in the end of the string. It is basic what you do in editor,
- end editing a textfield with pressing "enter", without creating a newline in the string
- ability to copy-pste or duplicate a row.
- maybe duplicate multiple rows and between editors?
- searching for some text and open the results in the treeview.
- detailed answer about what exactly is wrong in the file, in case of an error
- 
Thanks for possible attention.
I could participate in this project, best regards

**Comment by atenfyr** [comment]
> Duplicating rows would save my sanity right now. Are there any alternatives that can do it?

You can simply copy a row and paste it to the row below it.

**Comment by mczubaj** [comment]
> > Duplicating rows would save my sanity right now. Are there any alternatives that can do it?
> 
> You can simply copy a row and paste it to the row below it.

Yeah but that removes all the properties. D: You then have to copypaste them **one by one** (can't even select all of them at once) from the original row. Or is there a better way?

**Comment by atenfyr** [comment]
> Yeah but that removes all the properties. D: You then have to copypaste them **one by one** (can't even select all of them at once) from the original row. Or is there a better way?

No, if you select a row and copy the row, the property itself is copied to the clipboard as JSON, and pasting it will paste all data associated with that property, including e.g. data contained within structs and arrays. However, if you copy a struct, you need to execute View -> Recalculate Nodes to properly view sub-nodes in the tree view. If you have disabled the "Double click to edit" setting, you will need to re-enable it to copy rows. If you do not observe this behavior, this is a bug and it would be great if you could create a separate issue about it.

**Comment by atenfyr** [comment]
I'm going to close this issue, as all of the planned features suggested in this issue have been completed as of commit 1918ad0.

> * possibility to copy/paste text by ctrl c ctrl v via normal operating system buffer

This can be done by simply editing a cell and selecting the text/pasting into the cell.
> * please trim string input fields, because it may go wrong if you have spaces or newlines at the begin or in the end of the string. It is basic what you do in editor,

I've implemented this in commit 1918ad0 for the name and property fields. I don't believe it's appropriate to trim other fields, where there is a stronger expectation that the value being input is serialized directly.
> * end editing a textfield with pressing "enter", without creating a newline in the string

This seems to already be implemented. You need to use shift+enter to enter a new line into a string field.
> * ability to copy-pste or duplicate a row.
> * maybe duplicate multiple rows and between editors?

As of commit 186e0d6 (Sep 3, 2022), This can now done by ensuring the "Double click to edit" setting is disabled, selecting a row, and using CTRL+C or CTRL+V. The relevant property is copied to the clipboard as UAssetAPI JSON, and can be pasted to duplicate the row within the same or other open instances of UAssetGUI.
> * searching for some text and open the results in the treeview.

As of commit 2278ec1 (Sep 4, 2022), there now exists an Edit --> Find... menu (CTRL + F).
> * detailed answer about what exactly is wrong in the file, in case of an error

Most errors occur in UAssetAPI because there is some sort of deviation in serialization that led to some data being misinterpreted as having a different format than it really does, which leads to unexpected and undefined behavior, meaning that mostly useless error messages are thrown. There are many other error messages within UAssetAPI that are highly specific, but either way, error messages related to serialization I think aren't actually very useful for end users. I make an effort to provide useful pop-up information related to issues that end users can actually address when loading an asset in UAssetGUI, but when serialization fails because of a corrupt asset or because of a mistake or missing feature in UAssetAPI, there's not much that end users can do other than report the issue. That being said, it indeed would be useful to provide more useful contextual information within otherwise useless error messages to aid in development, but that will always be an ongoing process, and there will always be some confusing error messages.
